### PR TITLE
[FEA]  Migrate nvtext/edit_distance APIs to pylibcudf

### PR DESCRIPTION
**Pull request by Matt711**
## Description

Apart of #15162. This PR migrates `edit_distance.pxd` to pylibcudf
## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/pylibcudf/pylibcudf/tests/test_nvtext_edit_distance.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
```
Comment (Matt711) [inline python/pylibcudf/pylibcudf/tests/test_nvtext_edit_distance.py:1]: TODO: Use a fixture

File: python/pylibcudf/pylibcudf/__init__.pxd [hunk 2]
```diff
@@ -78,4 +79,5 @@ __all__ = [
     "transpose",
     "types",
     "unary",
+    "nvtext",
```
Comment (mroeschke) [inline python/pylibcudf/pylibcudf/__init__.pxd:82]: Nit: Could you keep list alphabetical as well?

File: python/pylibcudf/pylibcudf/__init__.py [hunk 3]
```diff
@@ -92,4 +93,5 @@
     "transpose",
     "types",
     "unary",
+    "nvtext",
```
Comment (mroeschke) [inline python/pylibcudf/pylibcudf/__init__.py:96]: Nit: Could you keep list alphabetical as well?

**Review by mroeschke — APPROVED** [review]
Modifications can be done in a follow up, but LGTM

File: python/pylibcudf/pylibcudf/__init__.pxd (hunk 2 above)
Comment (Matt711) [reply python/pylibcudf/pylibcudf/__init__.pxd:82]: Sure thing, will follow up

**Review by PointKernel — APPROVED** [review]

**Review by davidwendt — APPROVED** [review]
C++ approval

**Pull request merged**